shopify_python/google_styleguide.py

```diff
@@ -0,0 +1,158 @@
+        self.__dont_use_archaic_raise_syntax(node)
+
+    def __import_modules_only(self, node):  # type: (astroid.ImportFrom) -> None
+        """Use imports for packages and modules only."""
+
+        if hasattr(self.linter, 'config') and 'import-modules-only' in self.linter.config.disable:
+            return  # Skip if disable to avoid side-effects from importing modules
+
+        def can_import(module_name):
+            if sys.version_info >= (3, 4):
+                try:
+                    return bool(importlib.util.find_spec(module_name))
+                except AttributeError:
+                    return False
+            else:
+                try:
+                    importlib.import_module(module_name)
+                    return True
+                except ImportError:
+                    return False
+
+        if not node.level and node.modname != '__future__':
+            for name in node.names:
+                name, _ = name
+                parent_module = node.modname
+                child_module = '.'.join((node.modname, name))  # Rearrange "from x import y" as "import x.y"
+                if can_import(parent_module) and not can_import(child_module):
+                    self.add_message('import-modules-only', node=node)
+
+    def __import_full_path_only(self, node):  # type: (astroid.ImportFrom) -> None
+        """Import each module using the full pathname location of the module."""
+        if node.level:
+            self.add_message('import-full-path', node=node)
+
+    def __avoid_global_variables(self, node):  # type: (astroid.Assign) -> None
+        """Avoid global variables."""
+        if isinstance(node.parent, astroid.Module):
+            for target in node.targets:
+                if hasattr(target, 'elts'):
+                    for elt in target.elts:
+                        if elt.name != '__version__':
+                            self.add_message('global-variable', node=elt)
+                elif hasattr(target, 'name'):
+                    if target.name != '__version__':
+                        self.add_message('global-variable', node=target)
+
+    def __dont_use_archaic_raise_syntax(self, node):  # type: (astroid.Raise) -> None
+        """Don't use the two-argument form of raise or the string raise"""
+        children = list(node.get_children())
+        if len(children) > 1:
+            self.add_message('two-arg-exception', node=node)
+        elif len(children) == 1 and isinstance(children[0], six.string_types):
+            self.add_message('string-exception', node=node)
+
+    def __dont_catch_standard_error(self, node):  # type: (astroid.ExceptHandler) -> None
+        """
+        Never use catch-all 'except:' statements, or catch Exception or StandardError.
+
+        Pylint already handles bare-except and broad-except (for Exception).
+        """
+        if node.type.name == 'StandardError':
+            self.add_message('catch-standard-error', node=node)
```

Comment:
I think it's fine to catch `Exception` as long as you re-`raise` later on in the handler. You might also want to support the [exception chaining syntax](https://www.python.org/dev/peps/pep-0344/).